src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -459,12 +462,31 @@ private static int BioWrite(SafeBioHandle bio, byte[] buffer, int offset, int co
             return bytes;
         }
 
-        private static Ssl.SslErrorCode GetSslError(SafeSslHandle context, int result)
+        private static Ssl.SslErrorCode GetSslError(SafeSslHandle context, int result, out Exception innerError)
         {
+            ErrorInfo lastErrno = Sys.GetLastErrorInfo(); // cache it before we make more P/Invoke calls, just in case we need it
+
             Ssl.SslErrorCode retVal = Ssl.SslGetError(context, result);
-            if (retVal == Ssl.SslErrorCode.SSL_ERROR_SYSCALL)
+            switch (retVal)
             {
-                retVal = (Ssl.SslErrorCode)Crypto.ErrGetError();
+                case Ssl.SslErrorCode.SSL_ERROR_SYSCALL:
+                    // Some I/O error occurred
+                    innerError =
+                        Crypto.ErrPeekError() != 0 ? Crypto.CreateOpenSslCryptographicException() : // crypto error queue not empty
```

Comment:
Unrelated to this PR: Is there some plan to change this also to not use static exception creation methods?